swift-extensions/kingfisher/KFImageViewModelHandler.swift

```diff
@@ -0,0 +1,148 @@
+import Foundation
+import Kingfisher
+import TRIKOT_FRAMEWORK_NAME
+import Trikot_streams
+import UIKit
+
+public protocol KFImageUrlRequestModifierDelegate: AnyObject {
+    func requestModifier(for url: URL) -> ImageDownloadRequestModifier
+}
+
+public class KFImageViewModelHandler: ImageViewModelHandler {
+
+    public weak var delegate: KFImageUrlRequestModifierDelegate?
+    
+    public var isImageDependantOnViewSize = false
+    public var sizeMultiplier: CGFloat = UIScreen.main.scale
+
+    public init() {}
+
+    public func handleImage(imageViewModel: ImageViewModel?, on imageView: UIImageView) {
+        guard let imageViewModel = imageViewModel else {
+            imageView.unsubscribeFromAllPublisher()
+            return
+        }
+
+        imageView.image = nil
+        imageView.restoreContentMode()
+        
+        let imageFlowPublisher = imageViewModel.imageFlow(
+            width: Int32(imageView.frame.width * sizeMultiplier),
+            height: Int32(imageView.frame.height * sizeMultiplier)
+        )
+        
+        let cancellableManagerProvider = CancellableManagerProvider()
+        
+        observeImageFlow(
+            imageFlowPublisher,
+            cancellableManager: cancellableManagerProvider.cancelPreviousAndCreate(),
+            imageViewModel: imageViewModel,
+            imageView: imageView
+        )
+        
+        imageView.trikotInternalPublisherCancellableManager.add(cancellable: cancellableManagerProvider)
+    }
+    
+    private func observeImageFlow(
+        _ imageFlowPublisher: Publisher,
+        cancellableManager: CancellableManager,
+        imageViewModel: ImageViewModel,
+        imageView: UIImageView
+    ) {
+        let cancellableManagerProvider = CancellableManagerProvider()
+        cancellableManager.add(cancellable: cancellableManagerProvider)
+
+        imageView.observe(cancellableManager: cancellableManager, publisher: imageFlowPublisher) {[weak self] (imageFlow: ImageFlow) in
+            self?.doLoadImageFlow(
+                cancellableManager: cancellableManagerProvider.cancelPreviousAndCreate(),
+                imageViewModel: imageViewModel,
+                imageFlow: imageFlow,
+                imageView: imageView
+            )
+        }
+    }
+    
+    private func doLoadImageFlow(
+        cancellableManager: CancellableManager,
+        imageViewModel: ImageViewModel,
+        imageFlow: ImageFlow,
+        imageView: UIImageView
+    ) {
+        var unProcessedImage: UIImage?
+        
+        if let imageResource = imageFlow.imageResource {
```

Comment:
Maybe i'm too early in the morning 😂 but I do not understand why these `if let` fall thru.

if you have an `imageResource` or an `placeholderImageResource` shouldn't you return after setting the tintColor ?

Or we can really have all 3 cases in the same flow ? 
- placeholderImageResource
- imageResource
- url 

Thanks

Reply:
Actually, setting the `placeholderImageResource` before the download make sens 😅  could it be optional ? Some cases might want to set it only if the download fails